src/ensembl/src/shared/select/Select.tsx

```diff
@@ -0,0 +1,175 @@
+
+type SelectProps = OptionGroupsSelectProps & { placeholder: string };
+
+// TODO: think of a better name for this component?
+const ClosedSelect = (props: ClosedSelectProps) => {
+  const className = props.isOpen
+    ? styles.selectClosedInvisible
+    : styles.selectClosed;
+  return (
+    <span className={className} onClick={props.onClick}>
+      {props.selectedOption ? props.selectedOption.label : props.placeholder}
+      <SelectArrowhead />
+    </span>
+  );
+};
+
+const Select = (props: SelectProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedOption, optionGroups] = splitFromSelected(props.optionGroups);
+
+  // use ref to keep track of whether the element is focused
+  // (using ref makes this value available to the callback called from useEffect)
+  const focusRef = useRef(false);
+
+  useEffect(() => {
+    document.addEventListener('keydown', handleKeyPress);
+    return () => document.removeEventListener('keyup', handleKeyPress);
+  }, []);
+
+  const openPanel = () => {
+    setIsOpen(true);
+  };
+
+  const closePanel = () => {
+    setIsOpen(false);
+  };
+
+  const handleFocus = () => {
+    focusRef.current = true;
+  };
+
+  const handleBlur = () => {
+    focusRef.current = false;
+    if (isOpen) {
+      setIsOpen(false);
+    }
+  };
+
+  const handleKeyPress = (event: KeyboardEvent) => {
+    if (
+      !(
+        focusRef.current &&
+        [keyCodes.ENTER, keyCodes.ESC].includes(event.keyCode)
+      )
+    ) {
+      return;
+    }
+
+    if (event.keyCode === keyCodes.ENTER) {
+      setIsOpen(true);
+    } else if (event.keyCode === keyCodes.ESC) {
+      setIsOpen(false);
+    }
+  };
+
+  const handleSelect = (optionIndex: GroupedOptionIndex) => {
+    const [groupdIndex, itemIndex] = optionIndex;
+    const selectedOption = optionGroups[groupdIndex].options[itemIndex];
+    props.onSelect(selectedOption.value);
+    setIsOpen(false);
+  };
+
+  const headerText = selectedOption ? selectedOption.label : props.placeholder;
+  const className = classNames(styles.select, { [styles.selectOpen]: isOpen });
+
+  return (
+    <div
+      className={className}
+      tabIndex={0}
+      onFocus={handleFocus}
+      onBlur={handleBlur}
+    >
+      <ClosedSelect
+        isOpen={isOpen}
+        selectedOption={selectedOption}
+        onClick={openPanel}
+        placeholder={props.placeholder}
+      />
+      {isOpen && (
+        <SelectOptionsPanel
+          optionGroups={optionGroups}
+          header={headerText}
+          onSelect={handleSelect}
+          onClose={closePanel}
+        />
+      )}
+    </div>
+  );
+};
+
+Select.defaultProps = {
+  placeholder: 'Select'
+};
+
+// the purpose of the adapter is to unify props
+// to be consumed by the Select component
+const SelectAdapter = (props: SelectAdapterProps) => {
+  if ((props as OptionGroupsSelectProps).optionGroups) {
+    return <Select {...props as OptionGroupsSelectProps} />;
+  } else {
```

Comment:
The `else` part could just be put after the `if` without {}.

Reply:
Or use an early return without using `else`?

Reply:
@azangru Is this going to be changed or stay as it is?

Reply:
@azangru Sorry, I checked the wrong file commit. The latest commit has this changed.